hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java

```diff
@@ -303,7 +303,7 @@ protected synchronized void addStream(ChunkInfo chunkInfo) {
 
   protected ChunkInputStream createChunkInputStream(ChunkInfo chunkInfo) {
     return new ChunkInputStream(chunkInfo, blockID,
-        xceiverClientFactory, () -> pipeline, verifyChecksum, token);
+        xceiverClientFactory, () -> pipeline, verifyChecksum, () -> token);
```

Comment:
I may not understand the change -- why using a supplier here?  Since the taken is already created, passing `token` and pass `() -> token` are the same.

Reply:
If the token is updated, we want ChunkInputStream to use the new token.

Reply:
(At least, that's what I intend to achieve)

Reply:
I got it now.  The token will be updated form time to time.

Reply:
Should we need to make it `volatile`?

Reply:
I thought about making it (and `pipeline`) an `AtomicReference`.

Reply:
Yes, use `AtomicReference` is good.